### PR TITLE
Improve testing

### PR DESCRIPTION
**Pull request by julien-devatom**
### Description of change
- Refactor of the new tests
- Add tests that are going to be updated each time a new epoch is starting/finishing

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] `yarn lint` passes with this change
- [x] `yarn test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: test/marketsDistribution.test.ts [hunk 1]
```diff
@@ -21,3 +24,67 @@ describe("Markets distribution", () => {
...
+    Object.keys(onChainDistribution.proofs).forEach((onChainUser) => expect(newDistriUsers).toContain(onChainUser));
+  });
+  it("Should derive of 7e-18 only", () => {
+    const newDistriUsers = Object.entries(newDistribution.proofs);
+    Object.entries(onChainDistribution.proofs).forEach(([onChainUser, onChainDistrib]) => {
+      const [, compareTo] = newDistriUsers.find(([u]) => u === onChainUser)!;
+      // @ts-ignore unknown type
+      const compareToAmount = BigNumber.from(compareTo.amount);
+      // @ts-ignore unknown type
+      const onChainAmount = BigNumber.from(onChainDistrib.amount);
```
Comment (ghivert) [inline test/marketsDistribution.test.ts:46]: Can't the type be properly inferred?

File: test/marketsDistribution.test.ts [hunk 2]
```diff
@@ -21,3 +24,67 @@ describe("Markets distribution", () => {
...
+      Object.keys(onChainDistribution.proofs).forEach((onChainUser) => expect(newDistriUsers).toContain(onChainUser));
+    });
+    it("Should derive of 7e-18 only", () => {
+      const newDistriUsers = Object.entries(newDistribution.proofs);
+      Object.entries(onChainDistribution.proofs).forEach(([onChainUser, onChainDistrib]) => {
+        const [, compareTo] = newDistriUsers.find(([u]) => u === onChainUser)!;
+        // @ts-ignore unknown type
+        const compareToAmount = BigNumber.from(compareTo.amount);
+        // @ts-ignore unknown type
+        const onChainAmount = BigNumber.from(onChainDistrib.amount);
```
Comment (ghivert) [inline test/marketsDistribution.test.ts:73]: Same as above

File: test/marketsDistribution.test.ts [hunk 3]
```diff
@@ -21,3 +24,67 @@ describe("Markets distribution", () => {
...
+    const newDistribution = require("../distribution/proofs/proofs-2.json");
+
+    it("Should have the same number of users", () =>
+      expect(Object.keys(onChainDistribution.proofs).length).toEqual(Object.keys(newDistribution.proofs).length));
+
+    it("Should contains the same users", () => {
+      const newDistriUsers = Object.keys(newDistribution.proofs);
+      Object.keys(onChainDistribution.proofs).forEach((onChainUser) => expect(newDistriUsers).toContain(onChainUser));
+    });
+    it("Should derive of 7e-18 only", () => {
```
Comment (ghivert) [inline test/marketsDistribution.test.ts:66]: The test is a copy/paste of the test above, could you make a function of its body? (It's better for reviewers.)

**Review by ghivert — APPROVED** [review]

File: test/marketsDistribution.test.ts [hunk 4]
```diff
@@ -21,3 +24,41 @@ describe("Markets distribution", () => {
...
+
+  it("Should have the same number of users", () =>
+    expect(Object.keys(onChainDistribution.proofs).length).toEqual(Object.keys(newDistribution.proofs).length));
+
+  it("Should contains the same users", () => {
+    const newDistriUsers = Object.keys(newDistribution.proofs);
+    Object.keys(onChainDistribution.proofs).forEach((onChainUser) => expect(newDistriUsers).toContain(onChainUser));
+  });
+
+  it("Should derive of 7e-18 only", () => {
```
Comment (mlgarchery) [inline test/marketsDistribution.test.ts:40]: From where does  the 7 wei value come from ?

File: test/marketsEmission.test.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          expect(Object.keys(marketsEmissions)).toEqual(Object.keys(file.markets));
+        });
+        it(`Should have the correct supply rates computed for ${epochConfig.id}`, async () => {
+          Object.entries(marketsEmissions).forEach(([market, emission]) =>
+            expect(emission!.supplyRate.toString()).toEqual(file.markets[market].supplyRate)
+          );
+        });
+        it(`Should have the correct borrow rates computed for ${epochConfig.id}`, async () => {
+          Object.entries(marketsEmissions).forEach(([market, emission]) =>
+            expect(emission!.supplyRate.toString()).toEqual(file.markets[market].supplyRate)
```
Comment (mlgarchery) [inline test/marketsEmission.test.ts:70]:
```suggestion
            expect(emission!.borrowRate.toString()).toEqual(file.markets[market].borrowRate)
```

**Review by mlgarchery — CHANGES_REQUESTED** [review]

File: test/marketsDistribution.test.ts (hunk 4 above)
Comment (julien-devatom) [reply test/marketsDistribution.test.ts:40]: This is the biggest diff from prev script to the second one

**Review by mlgarchery — APPROVED** [review]

**Pull request merged**